### PR TITLE
replacing missing avatar photo with the team's github avatars

### PR DESCRIPTION
**Pull request by roosdebildt**

### CODE REVIEW

**Comment by emcoding** [comment]
@roosdebildt 
I tried to find the code you need in order to show a student's GH avatar.
Does this look useful to you? (instead of happyface.png)
``

I cut it out of a .slim and it is something I would put in a .html.erb div if we had those kind of things :-)
I really don't know yet if this user.avatar_url is accessible from your files, so if it is not working, it is me, not you. 
(I found it in Users show.html.slim file , under .profile ) 

Should it not work, than please could you test if this link does: 
`` 
to get your own avatar instead of the (other) happy face.

**Comment by roosdebildt** [comment]
I found that too, so there is a helper for user avatar. I'm now fiddling around with it to make it work :)

File: app/views/activities/_status_update.html.slim [hunk 1]
```diff
@@ -1,9 +1,10 @@
 li class="activity #{activity.kind}"
-  = image_tag(activity.img_url || '/images/screenshot-missing.png')
+  / = image_tag(activity.img_url || '/images/happyface.png')
+  / = image_tag(activity.img_url || '//avatars.githubusercontent.com/u/11146524')
+  = image_tag(activity.img_url || "#{user.avatar_url}")
```
Comment (andremedeiros) [inline app/views/activities/_status_update.html.slim]: Here you use

``` ruby
"#{user.avatar_url}"
```

Because the whole string is just the contents of the variable, it should just be:

``` ruby
user.avatar_url
```

File: app/views/activities/_feed_entry.html.slim [hunk 2]
```diff
@@ -1,10 +1,10 @@
 li class="activity #{activity.kind}"
   = link_to activity.source_url
-    = image_tag(activity.img_url || '/images/screenshot-missing.png')
-  = content_tag :h4, link_to(activity.title, activity.source_url, class: 'title')
+    - activity.students.each do |student|
```
Comment (roosdebildt) [inline app/views/activities/_feed_entry.html.slim]: Line 3 and 4 are not correct, they should be deleted again. Don't mind this!

**Comment by roosdebildt** [comment]
@gerwitz or @andremedeiros could you review if it's ready to be pushed to master and to upstream?

File: Gemfile [hunk 3]
```diff
@@ -54,7 +54,7 @@ end
 group :development do
   gem 'quiet_assets'
   gem 'better_errors'
-  # gem 'binding_of_caller'
+  gem 'binding_of_caller'
```
Comment (carpodaster) [inline Gemfile]: I wouldn't re-introduce this here as part of this PR

File: app/views/activities/_feed_entry.html.slim [hunk 4]
```diff
@@ -1,10 +1,14 @@
 li class="activity #{activity.kind}"
   = link_to activity.source_url
-    = image_tag(activity.img_url || '/images/screenshot-missing.png')
-  = content_tag :h4, link_to(activity.title, activity.source_url, class: 'title')
+  	- if activity.img_url.present?
```
Comment (carpodaster) [inline app/views/activities/_feed_entry.html.slim]: It would seem that you are using tabs instead of spaces. Could you configure your editor(s) to use 2 spaces for indentation?

File: app/views/activities/_status_update.html.slim [hunk 5]
```diff
@@ -1,9 +1,10 @@
 li class="activity #{activity.kind}"
-  = image_tag(activity.img_url || '/images/screenshot-missing.png')
-  = content_tag :h4, link_to(activity.title, status_update_path(activity)), class: 'title'
-  p.content= render_markdown(activity.content).html_safe
-  p.info
-    span.published_at = "at #{activity.published_at.to_formatted_s(:time)}".html_safe if activity.published_at
-    ' &nbsp;&mdash; by:
-    = link_to activity.team.display_name, activity.team
+	- activity.students.each do |student|
```
Comment (carpodaster) [inline app/views/activities/_status_update.html.slim]: See above: tabs vs. spaces

**Comment by yorickpeterse** [comment]
Before pushing this back to the original Rails Girls repo it might be a wise idea to rebase the commits so that any intermediate "oops" and "testing X" commits are gone, this keeps the history clean. This requires some extra Git fu so it might be a bit too much.

**Comment by carpodaster** [comment]
I'm very excited to see this implemented – well done :smile: 

One thing: It looks like this PR introduces different widths for the images for the activity overview, depending on whether it's a blog post screenshot ("source") or an activity image made from two gravatars. Then again, I'm terrible when it comes to reading CSS ;)

**Comment by roosdebildt** [comment]
@carpodaster you're right about that CSS thing! that's so that two avatars next to each other will have the same width as one screenshot and all the text will be aligned.

@YorickPeterse you mean squashing, right? maybe when the PR is ready one of the coaches can walk us through that :+1:

**Comment by carpodaster** [comment]
> Before pushing this back to the original Rails Girls repo it might be a wise idea to rebase the commits so that any intermediate "oops" and "testing X" commits are gone, this keeps the history clean. This requires some extra Git fu so it might be a bit too much.

@YorickPeterse oh yes, that would be "lecker", but it's perfectly fine to commit with all the oopses!

File: app/views/activities/_feed_entry.html.slim (hunk 4 above)
Comment (roosdebildt) [inline app/views/activities/_feed_entry.html.slim]: I just did that, but that breaks it locally... this is how it works locally for some reason. do you have an explanation for that?

File: app/views/activities/_status_update.html.slim (hunk 5 above)
Comment (roosdebildt) [inline app/views/activities/_status_update.html.slim]: See above.

File: Gemfile (hunk 3 above)
Comment (roosdebildt) [inline Gemfile]: Ok. Will take it out of the PR. @andremedeiros, where did we need this for again?

File: app/views/activities/_feed_entry.html.slim (hunk 4 above)
Comment (carpodaster) [inline app/views/activities/_feed_entry.html.slim]: slim / haml and the like are most picky when it comes to indentation. I can only think of spaces not adding up somewhere.

File: Gemfile (hunk 3 above)
Comment (emcoding) [inline Gemfile]: I suppose Andre did that for some debugging. @carpodaster : I enabled it as well, in a local debugging branch. Do you mean we shouldn't introduce it at all, or just not as a part of THIS particular PR?

Comment (carpodaster) [inline Gemfile]: @roosdebildt just add the comment again, re-bundle and make another commit. It's cool to have all the intermediary commits in this PR. RGSoC is about learning, it would make no sense if we didn't accept commits that bounce back and force or revert an "oops" ;)

Comment (carpodaster) [inline Gemfile]: @F3PiX Just not with this PR

File: app/views/activities/_feed_entry.html.slim (hunk 4 above)
Comment (roosdebildt) [inline app/views/activities/_feed_entry.html.slim]: ok, I'm looking at it now, I'll make it work! :)

Comment (carpodaster) [inline app/views/activities/_feed_entry.html.slim]: Yes please. As a rule of thumb, mixing tabs and spaces in a code-base means trouble. Ruby has settled on 2 spaces for indentation (as did Python, I think), JS is on 4 spaces and so every other language has their community best practices

**Comment by roosdebildt** [comment]
@carpodaster you think it's ready for upstream now? :)

File: app/models/activity.rb [hunk 6]
```diff
@@ -7,6 +7,8 @@ class Activity < ActiveRecord::Base
 
   validates :content, :title, :team, presence: { if: ->(act) { act.kind == 'status_update' } }
 
+  delegate :students, to: :team
```
Comment (carpodaster) [inline app/models/activity.rb:10]: Ideally, there would be a unit-test for this. It's a simple one, really, just to avoid regressions. But if you feel coping with rspec is too much at this time, we can skip that (or leave that for later)

File: app/views/activities/_feed_entry.html.slim [hunk 7]
```diff
@@ -1,10 +1,14 @@
 li class="activity #{activity.kind}"
   = link_to activity.source_url
-    = image_tag(activity.img_url || '/images/screenshot-missing.png')
-  = content_tag :h4, link_to(activity.title, activity.source_url, class: 'title')
+    - if activity.img_url.present?
+	    = image_tag(activity.img_url, class: "source")
```
Comment (carpodaster) [inline app/views/activities/_feed_entry.html.slim]: Looks like tabs still

**Comment by carpodaster** [comment]
@roosdebildt I've added two more comments. `app/views/activities/_feed_entry.html.slim` still uses tabs instead of spaces by the looks of it.

File: app/views/activities/_feed_entry.html.slim (hunk 7 above)
Comment (roosdebildt) [inline app/views/activities/_feed_entry.html.slim]: Crazy, you were right! I changed and doublechecked it now ;)

File: app/models/activity.rb (hunk 6 above)
Comment (roosdebildt) [inline app/models/activity.rb:10]: No skipping is for pussies! I'll look into how to do this, ask my genius back-end partner @F3PiX  and will get back to you :+1:

**Comment by yorickpeterse** [comment]
@roosdebildt 

> you mean squashing, right? maybe when the PR is ready one of the coaches can walk us through that

More or less, squashing usually means merging all commits into a single one. I was referring more to using `git rebase` to merge some of the commits together and delete others. This however is an advanced topic that's probably best explained in person or via a screencast of sorts.

**Comment by roosdebildt** [comment]
@YorickPeterse I used `git rebase` to rebase my branch with master. But I guess you're referring to something else? In that case, hell yes coaching session!

**Comment by yorickpeterse** [comment]
@roosdebildt Yes, `git rebase` is the one. I can explain how you'd use it in this case this evening during the drinkup.

**Comment by roosdebildt** [comment]
@YorickPeterse :+1: nice! maybe we can close the first issue all together then this evening!

File: app/models/activity.rb (hunk 6 above)
Comment (roosdebildt) [inline app/models/activity.rb:10]: @carpodaster, @F3PiX is just beginning to learn about testing, so we won't be able to do this any time soon. would it be ok if we skip it for now?

Comment (carpodaster) [inline app/models/activity.rb:10]: Yes, of course. :shipit:

Comment (emcoding) [inline app/models/activity.rb:10]: Phhhew. :sweat_smile:

**Comment by carpodaster** [comment]
I'm so excited to merrrrrge this! :confetti_ball:
